app-modules/portal/src/Http/Controllers/KnowledgeManagementPortal/KnowledgeManagementPortalCategoryController.php

```diff
@@ -74,6 +74,12 @@ public function show(KnowledgeBaseCategory $category): JsonResponse
             'articles' => $category->knowledgeBaseItems()
                 ->with('tags')
                 ->public()
+                ->when(request()->get('filter') === 'featured', function ($q) {
+                    $q->where('is_featured', true);
+                })
+                ->when(request()->get('filter') === 'most-viewed', function ($q) {
+                    $q->orderBy('portal_view_count', 'desc');
+                })
```

Comment:
For both of these functions lets change the variable name to `$query` and also typehint `Builder`